src/models/transformer.h

```diff
@@ -295,7 +295,8 @@ class Transformer : public EncoderOrDecoderBase {
       kh = cache_[prefix + "_keys"];                                                   // then return cached tensor
     }
     else {
-      auto Wk = graph_->param(prefix + "_Wk", {dimModel, dimModel}, inits::glorotUniform(true, true, depthScaling_ ? 1.f / sqrtf((float)depth_) : 1.f));
+      int dimKeys =  keys->shape()[-1];
```

Comment:
Add a comment when this will be different from `dimModel`. Concat factors only?

Reply:
done. Yes because when using concat factors the output of the encoder will be dimModel + dimFactEmb